src/NosCore.GameObject/Providers/InventoryService/InventoryService.cs

```diff
@@ -474,6 +474,40 @@ public InventoryItemInstance RemoveItemAmountFromInventory(short amount, Guid id
             return null;
         }
 
+        public List<InventoryItemInstance> RemoveItemAmountFromInventoryByVNum(short amount, short itemVNum)
+        {
+            List<InventoryItemInstance> result = new List<InventoryItemInstance>();
+            var stayAmount = amount;
+            foreach (var itemInstance in this.Select(s => s.Value)
+                .Where(w => w.ItemInstance.ItemVNum == itemVNum)
+                .OrderBy(o => o.ItemInstance.Amount))
+            {
+                if (stayAmount == 0)
+                {
+                    continue;
```

Comment:
Return a null value or a new list, using a continue still keeps the loop running, at this point, you can return "result" without problem, because it only relies on the parameters you passed to the method.

Also, you should use a "ushort" for the amount, so you don't have unexpected negative values that **might** cause a duplication failure

Reply:
Ok